### PR TITLE
Chore: Rename LookupTokenErr to AuthErr

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
With authn.Service and ClientTokenRotation flag we started to use LookupTokenErr from ReqContext as a way to pass authentication errors to auth middlewares so we can reject request from there.

It makes more sense to rename this to AuthErr.

Fixes #

**Special notes for your reviewer:**

### CODE REVIEW

File: pkg/services/contexthandler/contexthandler.go [hunk 1]
```diff
@@ -95,7 +92,6 @@ type ContextHandler struct {
 	oauthTokenService  oauthtoken.OAuthTokenService
 	features           *featuremgmt.FeatureManager
 	authnService       authn.Service
-	singleflight       *singleflight.Group
```
Comment (kalleep) [inline pkg/services/contexthandler/contexthandler.go:98]: Bonus: remove unused field

**Review by IevaVasiljeva — APPROVED** [review]

**Review by gamab — APPROVED** [review]

**Review by eleijonmarck — APPROVED** [review]

**Comment by sakjur** [comment]
fyi, I have a PR open that'd generalize error carrying on the request context a bit further (that I totally have not procrastinated merging for almost a month now), https://github.com/grafana/grafana/pull/64425

**Comment by kalleep** [comment]
@sakjur Yeah then I think it makes sense to not merge this one and wait for your pr to be merged and replace LookupTokenErr with the more generalized Error on request context, WDYT?

**Comment by sakjur** [comment]
@kalleep Yeah, if that works for y’all that makes sense to me

**Comment by kalleep** [comment]
Closing this in favor of https://github.com/grafana/grafana/pull/64425. When that one is merged we can remove LookupTokenErr and use Error instead

**Pull request closed without merging**